core/core.go

```diff
@@ -455,6 +464,11 @@ func (n *IpfsNode) startOnlineServicesWithHost(ctx context.Context, host p2phost
 		return err
 	}
 
+	// setup IPRS
+	kvs := iprsvs.NewKadValueStore(n.Repo.Datastore(), n.Routing)
+	n.Iprs = iprs.NewRecordSystem(kvs, size)
+	n.RecordFactory = iprsrec.NewRecordFactory(kvs)
```

Comment:
I think these details should be hidden behind the iprs facade.

Reply:
I refer to the record fatory.

Reply:
Yes I agree. At present the [code in namesys that increments the record sequence number](https://github.com/ipfs/go-ipfs/blob/master/namesys/publisher.go#L82) looks in the routing system's local datastore to avoid going out to the network, so I copied that same mechanism, but probably there is a better way of managing sequence numbers.

The Records are composed of a [RecordSigner](https://github.com/dirkmc/go-iprs/blob/master/record/record.go#L33) (eg private key or certificate) and [RecordValidity](https://github.com/dirkmc/go-iprs/blob/master/record/record.go#L20) (eg EOL or Time Range). These can be used in any permutation, and in future there will be more of them, eg [Signed, based on ancestry (chain)](https://github.com/ipfs/specs/tree/master/iprs#signed-based-on-ancestry-chain) so I thought it would get unwieldy to have a method to create every permutation in the interface itself.